### PR TITLE
add @types/passport-qq

### PR DESCRIPTION
**Pull request by bangbang93**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `pnpm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "@definitelytyped/dtslint/dt.json" }`, and no additional rules.
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

**Comment by DangerBotOSS** [comment]
## Formatting

The following files are not formatted:
1. types/passport-qq/package.json
1. types/passport-qq/index.d.ts
1. types/passport-qq/passport-qq-tests.ts

Consider running `pnpm dprint fmt` on these files to make review easier.
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 64e257a4aa3cc2ed45ea0ea39bbcc31fef7f4b08
</p>

File: types/passport-qq/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+export interface Options {
+    clientID: string;
+    clientSecret: string;
+    callbackURL: string;
+    authorizationURL?: string;
+    tokenURL?: string;
+    scopeSeparator?: string;
+}
+
+export type Callback = (err: Error | null, user: unknown) => void;
```
Comment (MysteryBlokHed) [inline types/passport-qq/index.d.ts:13]: If desirable, precise types for `user` could likely be added based on this: <https://github.com/qdsang/passport-qq/blob/23697a318dfac0d8ea0dd12b6ca66ed3efa05255/lib/passport-qq/strategy.js#L104-L109>

**Comment by DangerBotOSS** [comment]
## Formatting

The following files are not formatted:
1. types/passport-qq/index.d.ts

Consider running `pnpm dprint fmt` on these files to make review easier.
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 4e95beb77c4831abcd573d1dfe4dc9f6a1e76ea8
</p>

**Review by navya9singh — APPROVED** [review]

**Comment by bangbang93** [comment]
Ready to merge

**Pull request merged**